contracts/Controller.sol

```diff
@@ -124,38 +128,88 @@ contract Controller is Ownable {
     /**
      * @notice Return a vault balances, depend of the short option expiry
      * @dev if vault have no short option or issued option is not expired yet, return vault, else get excess margin and return it as collateral amount inside Vault struct.
-     * @param _owner vault owner.
+     * @param _owner account owner.
      * @param _vaultId vault.
      * @return Vault struct
      */
-    // function getVaultBalances(address _owner, uint256 _vaultId) external view returns (MarginAccount.Vault memory) {
-    // }
+    function getVaultBalances(address _owner, uint256 _vaultId) external view returns (MarginAccount.Vault memory) {
+        MarginAccount.Vault memory vault = getVault(_owner, _vaultId);
+
+        // if there is no minted option or option not expired yet
+        if ((vault.shortOtokens.length == 0) || (!isExpired(vault.shortOtokens[0]))) return vault;
+
+        // if there's short and it's expired
+        address calculatorModule = AddressBookInterface(addressBook).getMarginCalculator();
+        MarginCalculatorInterface calculator = MarginCalculatorInterface(calculatorModule);
+
+        (uint256 netValue, ) = calculator.getExcessMargin(vault, vault.shortOtokens[0]);
+        vault.collateralAmounts[0] = netValue;
```

Comment:
You should check if the vault has any collateral and return the min(vault.collateral, netValue)

Reply:
https://docs.google.com/document/d/13xDJoPqfR6H3N_AMdBuSV8t20pEGiUqo4kCOJkBEV3o/edit?disco=AAAAKICQk3k see

Reply:
Okay got that, but in the spec of `getVaultBalances()` it just say:
 ```
After Expiry:
CollateralAmounts = Controller.getExcessMargin(vault, vault.short.collateral);
Return vault {long, short, collateral: collateralAmount}
```
We should update that.